ring/ingester_lifecycle.go

```diff
@@ -16,125 +16,212 @@
 package ring
 
 import (
-	"encoding/json"
 	"fmt"
-	"hash/fnv"
 	"math/rand"
 	"net"
 	"os"
+	"sort"
 	"sync"
 	"time"
 
 	"github.com/prometheus/common/log"
 )
 
 const (
-	infName = "eth0"
+	infName           = "eth0"
+	ring              = "ring"
```

Comment:
Rather than calling it `ring` maybe name it after what it's for / how it's used?